modules/EED_SquareOnsiteOAuth.module.php

```diff
@@ -96,11 +96,26 @@ public static function requestAccess()
             // Not it. Ignore it.
             return;
         }
+
+        // Get the payment method.
+        $square_pm = EEM_Payment_Method::instance()->get_one_by_slug(sanitize_key($_GET['square_slug']));
+        if (! $square_pm instanceof EE_Payment_Method) {
+            EED_SquareOnsiteOAuth::closeOauthWindow(
+                esc_html__(
+                    'Could not specify the payment method !',
+                    'event_espresso'
+                )
+            );
+        }
+
         // Check that we have all the required parameters and the nonce is ok.
         if (! wp_verify_nonce($_GET['nonce'], 'eea_square_grab_access_token')) {
-            // This is an error. Close the window.
-            EED_SquareOnsiteOAuth::closeOauthWindow(esc_html__('Nonce fail!', 'event_espresso'));
+            // This is an error. Log and close the window.
+            $err_msg = esc_html__('Nonce fail!', 'event_espresso');
+            EED_SquareOnsiteOAuth::errorLogAndExit($square_pm, $err_msg, [], true, true);
         }
+        // log the incoming data, don't exit
+        EED_SquareOnsiteOAuth::errorLogAndExit($square_pm, 'Request access request response', $_GET, false);
```

Comment:
do we need i18n for log messages? what have we done for other PMs?

```suggestion
        EED_SquareOnsiteOAuth::errorLogAndExit(
                $square_pm,
                esc_html__('Request access request response', 'event_espresso')
                $_GET, 
                false
        );
```